### PR TITLE
Fix a Clippy warning

### PR DESCRIPTION
**Pull request by Thomasdezeeuw**

### CODE REVIEW

**Comment by dtacalau** [comment]
This change was also causing #1220 CI to fail. I've reverted it from #1220 and it passed.

**Comment by kleimkuhler** [comment]
The test no longer needs to wrap `waker` in an `Arc` with this change.

@dtacalau I'm starting to think this test may just be flaky and that adding back the `.clone()` just happened to hit trigger a successful run..

File: tests/regressions.rs [hunk 1]
```diff
@@ -60,7 +60,6 @@ fn issue_1205() {
     let (mut poll, mut events) = init_with_poll();
 
     let waker = Arc::new(Waker::new(poll.registry(), WAKE_TOKEN).unwrap());
```
Comment (kleimkuhler) [inline tests/regressions.rs]: `waker` doesn't need to be wrapped in an `Arc` anymore.

Comment (Thomasdezeeuw) [reply tests/regressions.rs]: I know but that was part of the regression test so I just left it.

File: tests/regressions.rs [hunk 2]
```diff
@@ -79,7 +78,7 @@ fn issue_1205() {
     // spawn a waker thread to wake the poll call below
     let handle = thread::spawn(move || {
         thread::sleep(Duration::from_millis(500));
-        waker1.wake().expect("unable to wake");
+        waker.wake().expect("unable to wake");
```
Comment (Thomasdezeeuw) [inline tests/regressions.rs]: I just realised this change drops the `waker` here and than `Poll` will never get a wakup event (see the `Waker` docs). So this change is invalid.

I'll send a change to ignore the Clippy warning.

**Pull request closed without merging**